### ISSUE TITLE
Reproducing the baseline results (ABMIL, DTFD etc.)

### ISSUE THREAD

**Issue by undercutspiky** [ChongQingNoSubway/DGR-MIL#5]
Hi,

For C16 dataset, the paper says

> For the CAMELYON16, the training set is further divided into training and validation sets with a 9:1 ratio

But I couldn't find the script used to divide the training set into train and val sets.
I was also wondering what strategy did you use to pick the ABMIL (and other baseline) models for reporting numbers.
It'd be of great help to me if you could help me figure out some details regarding CAMELYON16 experiments:

1. Did you split the training set into train & val only exactly once and then used that fixed split for all the methods (ABMIL, DTFD...DGR-MIL)?
2. For each method did you pick the best checkpoint based on accuracy on the val set and then reported the accuracy on the test set?
3. In your code it seems like you fix the random seed. Did you have to find a good random seed to pick for each method individually or do you simply set it to 0 for all the methods?
4. Do you accumulate gradients for any of the methods for any number of steps? From the paper and the code it seems like you don't but since some people do, I just wanna verify.

So far I've been unable to get consistent 80+ balanced accuracy using ABMIL on C16 (and 85+ for DTFD) but I'm not fixing a random seed so my results vary quite a bit for ABMIL...while they are stable but low for DTFD.

Thank you.

**Comment by undercutspiky** [comment]
**Update:** I was able to get good enough results with ABMIL by using your architecture (thanks for the this codebase) but I'm still struggling with DTFD. So, the questions above are more relevant for DTFD now. I'm yet to try out TransMIL.

**Comment by ChongQingNoSubway** [comment]
Thank you very much for your interest in our work, and apologies for the delayed reply as we have been busy with other projects recently.

1. Yes, we split the training set only once, and all methods use the same split. We used StratifiedKFold from sklearn for this.

2. Yes, we selected the best checkpoint based on performance on the validation set.

3. Yes, for most methods that can be reproduced reliably, we did not tune the random seed. However, for some methods that often fail to train properly (like DTFD, transmil sometimes), we had to adjust the random seed.

4. We did not use gradient accumulation. However, for DTFD, if you are using a newer version of PyTorch, you might need to change the graph parameter in backward(), which can cause performance degradation. The best solution is to use the environment provided in the official DTFD GitHub repository.

You could also try creating a new train/validation split. I hope this helps!

**Comment by undercutspiky** [comment]
Thank you for answering all the questions.
It's interesting that the random seed needs to be tuned for some methods but not for others.
